### PR TITLE
Prior specification and sign corrections

### PR DESCRIPTION
**Pull request by infinitron**
This PR extends the current interface to fully specify the prior on smoothing parameters. It also fixes signs in the prior's expression used while computing the log-posterior.

### CODE REVIEW

**Comment by vkashyap** [comment]
Thanks for the fixes, Karthik!

File: lira/src/bayes-image-analysis.c [hunk 1]
```diff
@@ -599,8 +613,8 @@ void allocate_memory(psfType *psf, expmapType *expmap, cntType *obs,
     Rprintf("Gamma(%f, %f).\n", ms->ttlcnt_pr, ms->ttlcnt_exp);
 
     ms->fit_al = 1;                 /**** fit or fix alpha ****/
-    ms->al_kap1 = 0.0;              /**** the prior for alpha ****/
-    ms->al_kap3 = 3.0;
+    ms->al_kap1 = ms->al_kap1 == -42 ? 0.0 : ms->al_kap1;              /**** the prior for alpha ****/
+    ms->al_kap3 = ms->al_kap3 == -42 ? 3.0 : ms->al_kap3;
```
Comment (anetasie) [inline lira/src/bayes-image-analysis.c:617]: I guess I don't quite understand -42 in the above statements

File: lira/src/bayes-image-analysis.c [hunk 2]
```diff
@@ -1643,8 +1657,8 @@ double lpost_alpha(double alpha, /* evaluate at alpha */
   } /* loop over rows of ms.ag[level] */
 
   /****************  ADD THE LOG PRIOR *************/
-  logpost -= (   ms->al_kap1 *log(alpha) 
-	       + ms->al_kap2 * pow( alpha, ms->al_kap3 ) );
+    logpost -= ( -ms->al_kap1 * log(ms->al_kap2)  +  ms->al_kap1 * log(alpha) 
+	        + ms->al_kap2 * pow( alpha, ms->al_kap3 ) );
```
Comment (anetasie) [inline lira/src/bayes-image-analysis.c:1661]: @infinitron I am wondering if requiring `al_kap1 < 0` would be enough to deal with the sign?

**Comment by anetasie** [comment]
Thanks! I just left two comments/questions

File: lira/src/bayes-image-analysis.c (hunk 2 above)
Comment (infinitron) [reply lira/src/bayes-image-analysis.c:1661]: Yes, that would work too. But I realize I missed one more sign in the second term in line# 1660. It should be negative as well.

File: lira/src/bayes-image-analysis.c (hunk 1 above)
Comment (infinitron) [reply lira/src/bayes-image-analysis.c:617]: These parameters are initialized and set to defaults (if not specified by the user) in two different functions. So I used a random value to indicate if they are not set.  I raised a new [PR](https://github.com/astrostat/LIRA/pull/5) with a name to this value.

**Pull request merged**